### DISCUSSION TITLE
Question: Does this supports 2.0 Private Servers?

### DISCUSSION THREAD

**Discussion by zero-cache** [zmxhawrhbg/gdrpc#4] [Q&A]
Hello! Any chance that this will work on my 2.0 Private Server?

Any help would be appreciated

**Comment by qimiko** [comment]
for a 2.0 private server, you would have to modify the addresses found in this array in `game_hooks.cpp`
https://github.com/zmxhawrhbg/gdrpc/blob/25cffa74b22e907ae4c83087ccf52003aab0e240/src/game_hooks.cpp#L198-L220
you might also have to edit some stuff in `gdapi.cpp` like removing the demon difficulty reading
https://github.com/zmxhawrhbg/gdrpc/blob/25cffa74b22e907ae4c83087ccf52003aab0e240/src/gdapi.cpp#L323-L326
there's also a chance that the way that gjgamelevels work internally changed, which would require some addresses to also be changed to reflect the 2.0 state 
however, this will not work unmodified on any non 2.1 client, including 2.0

i had some free time so here is an example port to the 1.9 GDPS client:
https://github.com/kyurime/gdrpc/commit/3f45533f70575f080e2742c535d1c399693c3ac7

(i will likely pin this issue if enough people ask for non 2.1 versions)

**Comment by qimiko** [comment]
it's been a month since this was opened so i'm going to close it, if there are any further questions feel free to reopen this
